File: packages/silo-client/src/cache/CacheStatistics.ts
/** A snapshot across this client's cache policies. Clear and explicit invalidation preserve counters. */
export class CacheStatistics {
  constructor(
    readonly hits: number,
    readonly misses: number,
    readonly evictions: number,
    readonly size: number,
  ) {}

  requests(): number {
    return this.hits + this.misses;
  }

  hitRate(): number {
    return this.requests() === 0 ? 0 : this.hits / this.requests();
  }
}

File: packages/silo-client-dart/lib/src/cache/cache_statistics.dart
/// A snapshot across every store in one cache. Clearing keeps the counters.
final class CacheStatistics {
  const CacheStatistics(this.hits, this.misses, this.evictions, this.size);

  final int hits;
  final int misses;
  final int evictions;
  final int size;

  int get requests => hits + misses;

  double get hitRate => requests == 0 ? 0 : hits / requests;
}
